compiler/src/iree/compiler/DispatchCreation/SetEncoding.cpp

```diff
@@ -306,10 +306,226 @@ struct FoldFillWithSetEncoding final
+
+  // Compute the result types of the new dispatch.
+  auto newResultType = operandType.cloneWithEncoding(encoding);
+  auto newResultTypes = llvm::to_vector(producerDispatch->getResultTypes());
+  newResultTypes[resultNumber] = newResultType;
+
+  // Compute the result dynamic dims.
+  SmallVector<OpFoldResult> operandDims =
+      tensor::getMixedSizes(rewriter, loc, operand);
+  SmallVector<Value> operandDynamicDims;
+  std::tie(std::ignore, operandDynamicDims) = decomposeMixedValues(operandDims);
+
+  SmallVector<Value> newResultDynamicDims;
+  for (OpResult producerDispatchResult : producerDispatch.getResults()) {
+    if (producerDispatchResult != producerValue) {
+      llvm::append_range(newResultDynamicDims,
+                         producerDispatch.getResultDynamicDims(
+                             producerDispatchResult.getResultNumber()));
+      continue;
+    }
+    llvm::append_range(newResultDynamicDims, operandDynamicDims);
+  }
+
+  auto newDispatchOp = rewriter.create<IREE::Flow::DispatchRegionOp>(
+      producerDispatch->getLoc(), newResultTypes, newResultDynamicDims,
+      producerDispatch.getWorkload());
+
+  // Move over the body of the old dispatch.
+  Region &newBody = newDispatchOp.getBody();
+  Region &producerDispatchBody = producerDispatch.getBody();
+  rewriter.cloneRegionBefore(producerDispatchBody, newBody, newBody.begin());
+
+  // Move over the slice operations if needed.
+  Region &producerWorkgroupCountBody = producerDispatch.getWorkgroupCount();
+  if (!producerWorkgroupCountBody.empty()) {
+    Region &newWorkgroupCountBody = newDispatchOp.getWorkgroupCount();
+    rewriter.cloneRegionBefore(producerWorkgroupCountBody,
+                               newWorkgroupCountBody,
+                               newWorkgroupCountBody.begin());
+  }
+
+  // Clone the operation chain.
+  IRMapping map;
+  auto returnOp = cast<IREE::Flow::ReturnOp>(newBody.front().getTerminator());
+  Value yieldedVal = returnOp.getOperand(resultNumber);
+
+  OpBuilder::InsertionGuard g(rewriter);
+  rewriter.setInsertionPoint(returnOp);
+  map.map(producerValue, yieldedVal);
+
+  for (Operation *op : llvm::reverse(opChain)) {
+    rewriter.clone(*op, map);
+  }
+  // Find the new value to yield
+  Value newYieldedVal = map.lookup(operand);
+  auto encodingOp = rewriter.create<IREE::Encoding::SetEncodingOp>(
+      returnOp->getLoc(), newResultType, newYieldedVal);
+  rewriter.modifyOpInPlace(
+      returnOp, [&]() { returnOp.setOperand(resultNumber, encodingOp); });
+
+  return PaddedValue{newDispatchOp->getResult(resultNumber),
+                     operandDynamicDims};
+}
+
+// For a given operation, pad the operands corresponding to `operandNums`.
+static SmallVector<unsigned> padOperandsOfOp(RewriterBase &rewriter,
+                                             Operation *op,
+                                             ArrayRef<unsigned> operandNums) {
+  // 1. Check that the operation is within a dispatch region
```

Comment:
style nit 1: prefer not using numbers. They always end up wrong when code is moved around or when code is inserted/removed. And there are no other numbers in comments.
style nit 2: Missing a period at the end.

optional nit: We can be more explicit instead of saying "Check". It saves me one step of translation between the comment and the code. It could just be me, so I marked it optional.

```suggestion
  // Do not pad the operands if the operation is not within a dispatch region.
```